src/Core/Resolvers/MsSqlQueryExecutor.cs

```diff
@@ -86,17 +101,22 @@ public override SqlConnectionStringBuilder ConnectionStringBuilder
         /// provided in the runtime configuration.
         /// </summary>
         /// <param name="conn">The supplied connection to modify for managed identity access.</param>
-        public override async Task SetManagedIdentityAccessTokenIfAnyAsync(DbConnection conn)
+        public override async Task SetManagedIdentityAccessTokenIfAnyAsync(DbConnection conn, string? datasourceName = null)
         {
+            if (string.IsNullOrEmpty(datasourceName))
+            {
+                datasourceName = ConfigProvider.GetConfig().DefaultDBName;
+            }
             // Only attempt to get the access token if the connection string is in the appropriate format
-            if (_attemptToSetAccessToken)
+            // using default for first db - maintaining backward compatibility for single db scenario.
```

Comment:
The comment makes more sense before line 108?

Reply:
fixed.